membership-attribute-service/app/controllers/HealthCheckController.scala

```diff
@@ -16,14 +17,25 @@ class BoolTest(name: String, exec: () => Boolean) extends Test {
   override def ok = exec()
 }
 
-class HealthCheckController extends Results {
-  lazy val zuora = NormalTouchpointComponents.zuoraService
-  lazy val salesforce = NormalTouchpointComponents.contactRepo.salesforce
+class HealthCheckController extends Results with LazyLogging {
 
-  val tests: Seq[Test] = Seq(
-    new BoolTest("ZuoraPing", () => zuora.lastPingTimeWithin(2.minutes)),
-    new BoolTest("Salesforce", () => salesforce.isAuthenticated)
-  )
+  private lazy val wrappedZuoraService = new HealthCheckableService {
+    private val duration = 2.minutes
+    override def serviceName = "ZuoraService"
+    override def checkHealth: Boolean =
+      if (zuoraService.lastPingTimeWithin(duration)) {
+        true
+      } else {
+        logger.error(s"${this.serviceName} has not responded within the last ${duration.getMinutes} minutes.")
+        // Return true rather than false, because we don't want Zuora to fail the healthcheck.
+        // However we do want to log outages because we have an SLA from Zuora where we can claim compensation.
+        true
+      }
+  }
+
+  private lazy val services = Set(wrappedZuoraService, salesforceService, featureToggleService, attrService, behaviourService)
```

Comment:
I know it's not specifically part of the retro action, but do we want Salesforce auth on the health check either?

Reply:
I would also question whether the behaviourService is really essential functionality for this app